libp2p/routing_record.nim

```diff
@@ -77,3 +84,42 @@ proc init*(T: typedesc[PeerRecord],
     seqNo: seqNo,
     addresses: addresses.mapIt(AddressInfo(address: it))
   )
+
+
+## Functions related to signed peer records
+
+proc init*(T: typedesc[Envelope],
+           privateKey: PrivateKey,
+           peerRecord: PeerRecord): Result[Envelope, CryptoError] =
+  
+  ## Init a signed envelope wrapping a peer record
+
+  let envelope = ? Envelope.init(privateKey,
+                                 EnvelopePayloadType,
+                                 peerRecord.encode(),
+                                 $EnvelopeDomain)
+  
+  ok(envelope)
+
+proc init*(T: typedesc[Envelope],
+           peerId: PeerId,
+           addresses: seq[MultiAddress],
+           privateKey: PrivateKey): Result[Envelope, CryptoError] =
+  ## Creates a signed peer record for this peer:
+  ## a peer routing record according to https://github.com/libp2p/specs/blob/500a7906dd7dd8f64e0af38de010ef7551fd61b6/RFC/0003-routing-records.md
+  ## in a signed envelope according to https://github.com/libp2p/specs/blob/500a7906dd7dd8f64e0af38de010ef7551fd61b6/RFC/0002-signed-envelopes.md
+  
+  # First create a peer record from the peer info
+  let peerRecord = PeerRecord.init(peerId,
+                                   getTime().toUnix().uint64, # This currently follows the recommended implementation, using unix epoch as seq no.
+                                   addresses)
+
+  let envelope = ? Envelope.init(privateKey,
+                                 peerRecord)
+  
+  ok(envelope)
+
+proc getField*(pb: ProtoBuffer, field: int,
+               value: var Envelope): ProtoResult[bool] {.
```

Comment:
We should make the RoutingRecord explicit, since there may be multiple envelope payload type in the future.

So getSignedPeerRecord?